portal/eproms_substudy_tailored_content/src/js/Utility.js

```diff
@@ -0,0 +1,113 @@
+export function getWrapperJS (el) {
+    if (!el) return;
+    let jsSet = document.querySelectorAll(`${el} script`); 
+    jsSet.forEach(item => {
+        var d=document, g=d.createElement("script"), b=d.getElementsByTagName("body")[0];
+        if (!item.getAttribute("src")) {
+            g.innerHTML = item.innerHTML;
+            b.appendChild(g);
+            return true;
+        }
+        let props = {
+            type: "text/javascript",
+            async: true,
+            defer: true,
+            src: item.getAttribute("src")
+        };
+        for (const [key, value] of Object.entries(props)) {
+            g[key] = value;
+        }
+        b.appendChild(g);
+    });
+};
+export function sendRequest (url, params) {
+
+    params = params || {};
+    // Return a new promise.
+    return new Promise(function(resolve, reject) {
+      // Do the usual XHR stuff
+      var req = new XMLHttpRequest();
+      req.open('GET', url);
+      req.onload = function() {
+        // This is called even on 404 etc
+        // so check the status
+        if (req.status == 200) {
+          // Resolve the promise with the response text
+          resolve(req.response);
+        }
+        else {
+          // Otherwise reject with the status text
+          // which will hopefully be a meaningful error
```

Comment:
Would it help identify clients that didn't set a meaningful error, by looking for valid data and adorning if not?

Reply:
yes, meaningful error(s) will be set (in the Base.js code) when no valid data is provided.